script/pathfinder.lua

```diff
@@ -0,0 +1,293 @@
+            -- elseif request.attempt == 3 then
+            --     game.print('Attempt 3')
+            -- elseif request.attempt == 4 then
+            --     game.print('Attempt 4')
+            -- elseif request.attempt == 5 then
+            --     game.print('Attempt 5')
+            -- elseif request.attempt == 6 then
+            --     game.print('Attempt 6')
+            -- end
+            if clean_linear_path_enabled then
+                path = pathfinder.clean_linear_path(path)
+            end
+            if clean_path_steps_enabled then
+                path = pathfinder.clean_path_steps(path, clean_path_steps_distance)
+            end
+            if request.landfill_job then
+                table.insert(path, {position = {x = request.initial_target.x, y = request.initial_target.y}}) -- add the desired destination after the walkable path
+            end
+            -- table.insert(path, {position = {x = request.initial_target.x, y = request.initial_target.y}}) -- add the desired destination after the walkable path
+            -- we currently cannot do this as the job condition will not match. Additionally, if we account for that it could result in a waypoint loop with FNCP.
+            -- possibly the poisition_done condition could be changed to check for proximity to both initial and updated destination.
+            if clean_path_steps_enabled then
+                path = pathfinder.clean_path_steps(path, 2.5)
+            end
+            if request.job then
+                global.job_bundles[request.job.bundle_index][1]["path_active"] = true
+            end
+            pathfinder.set_autopilot(request.unit, path)
+        end
+    end
+    global.pathfinder_requests[event.id] = nil
+end
+
+-------------------------------------------------------------------------------
+--  Utility
+-------------------------------------------------------------------------------
+
+---@param unit LuaEntity
+---@param path PathfinderWaypoint[]
+function pathfinder.set_autopilot(unit, path) -- set path
+    if unit and unit.valid then
+        unit.autopilot_destination = nil
+        for i, waypoint in ipairs(path) do
+            unit.add_autopilot_destination(waypoint.position)
+        end
+    end
+end
+
+---@param surface LuaSurface
+---@param position MapPosition
+---@return MapPosition?
+function pathfinder.find_non_colliding_position(surface, position, job) -- find a position to stand / walk to
+    local new_position
+    local bb
+    local params = {
+        {size = 12, radius = 64},
+        {size = 5, radius = 16},
+        {size = 5, radius = 8},
+        {size = 1, radius = 12},
+        {size = 1, radius = 5}
+        }
+    for _, param in pairs(params) do
+        new_position = surface.find_non_colliding_position("constructron_pathing_proxy_" .. param.size, position, param.radius, non_colliding_position_accuracy, false)
+        if new_position then
+            bb = param
+        end
```

Comment:
this seems like dead code? what purpose does this serve? looks like `bb` is never read, only written to.